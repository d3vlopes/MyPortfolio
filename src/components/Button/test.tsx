import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'
import { Github } from '@styled-icons/boxicons-logos'

import Button from '.'

const { getByRole, getByTestId } = screen

describe('<Button />', () => {
  it('should render by default', () => {
    renderWithTheme(
      <Button variant="primary" size="medium">
        Button
      </Button>,
    )

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'background-color': '#6D59A8',
      'font-size': '2.2rem',
      width: '21.5rem',
      height: '6.2rem',
    })
  })

  it('should render the variant secondary ', () => {
    renderWithTheme(<Button variant="secondary">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'background-color': '#83A848',
    })
  })

  it('should render the variant outline ', () => {
    renderWithTheme(<Button variant="outline">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      border: '1px solid #DEE7EA',
      'background-color': 'transparent',
    })
  })

  it('should render expand version', () => {
    renderWithTheme(<Button expand>Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      width: '31.5rem',
      height: '6.2rem',
    })
  })

  it('should render small size', () => {
    renderWithTheme(<Button size="small">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      'font-size': '1.8rem',
      width: '17.5rem',
      height: '5rem',
    })
  })

  it('should render large size', () => {
    renderWithTheme(<Button size="large">Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      'font-size': '2.6rem',
      width: '28.8rem',
      height: '7.6rem',
    })
  })

  it('should render an icon version', () => {
    renderWithTheme(
      <Button icon={<Github data-testid="icon" />}>Button</Button>,
    )

    const button = getByRole('button', { name: /button/i })
    const icon = getByTestId('icon')

    expect(button).toBeInTheDocument()
    expect(icon).toBeInTheDocument()
  })

  it('should render Button as a link', () => {
    renderWithTheme(
      <Button as="a" href="/link">
        Button
      </Button>,
    )

    expect(screen.getByRole('link', { name: /button/i })).toHaveAttribute(
      'href',
      '/link',
    )
  })

  it('should up while hover', async () => {
    renderWithTheme(<Button hoverUp>Button</Button>)

    const button = getByRole('button', { name: /button/i })

    expect(button).toHaveStyleRule('transform', 'translateY(-0.5rem)', {
      modifier: ':hover',
    })
  })

  it('should match snapshot', () => {
    const { container } = renderWithTheme(<Button>Button</Button>)

    expect(container.firstChild).toMatchSnapshot()
  })
})
