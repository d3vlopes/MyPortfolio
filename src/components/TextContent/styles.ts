import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.large};
    line-height: 180%;
    color: ${theme.colors.complementary.main};

    ${media.lessThan('medium')`
      font-size: ${theme.font.sizes.xsmall};
    `}

    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      font-family: ${theme.font.heading};
      color: ${theme.colors.text.light};
      margin: ${theme.spacings.xsmall} 0;
    }

    p {
      margin-bottom: ${theme.spacings.xsmall};
      max-width: 59.7rem;
    }

    a,
    a:visited,
    a:link {
      color: ${theme.colors.primary.main};
      text-decoration: none;
      transition: all ${theme.transition.default};
    }

    a:hover {
      color: ${theme.colors.primary.dark};

      &:hover > img {
        filter: brightness(50%);
      }
    }

    img {
      max-width: 100%;
      transition: all ${theme.transition.default};

      &:hover {
        transform: scale(1.2);
      }
    }

    .image {
      //background: ${theme.colors.bg.secondary.main};
      line-height: 0;
      margin: ${theme.spacings.small} 0;
    }

    .image figcaption {
      font-size: ${theme.font.sizes.xsmall};
      padding: ${theme.spacings.xsmall};
      text-align: center;
      line-height: 1.3;
    }

    .image-half {
      max-width: 50%;
    }

    .image-style-side {
      max-width: 50%;
      float: right;
      margin: ${theme.spacings.small} 0;

      ${media.lessThan('medium')`
      & {
        max-width: 100%;
        float: none;
        }
      `}
    }

    ul,
    ol {
      padding: 0 ${theme.spacings.small} ${theme.spacings.xsmall};
    }

    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.bg.secondary.main};
    }

    code {
      font-family: monospace;
      color: ${theme.colors.complementary.main};
      font-size: ${theme.font.sizes.xsmall};
      background: ${theme.colors.bg.secondary.dark};
      padding: 0.2rem;
      margin: 0.2rem;
    }

    pre {
      background: ${theme.colors.bg.secondary.dark};
      padding: ${theme.spacings.small};
      font-family: monospace;
      color: ${theme.colors.text.light};
      margin: ${theme.spacings.small} 0;
      width: 100%;
      overflow-x: auto;
      font-size: ${theme.font.sizes.xsmall};
    }

    pre code {
      color: inherit;
      background: inherit;
    }

    .table {
      width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }

    table {
      width: 100%;
      border-collapse: collapse;
      margin: ${theme.spacings.small} 0;
    }

    table td,
    table th {
      padding: ${theme.spacings.xsmall};
      border-bottom: 0.1rem solid ${theme.colors.bg.secondary.main};
    }

    blockquote {
      border-left: 0.5rem solid ${theme.colors.primary.main};
      color: ${theme.colors.complementary.main};
      /* filter: brightness(80%); */
      padding-left: ${theme.spacings.small};
      margin: ${theme.spacings.small};
    }
  `}
`
