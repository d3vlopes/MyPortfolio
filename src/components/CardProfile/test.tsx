import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import CardProfile, { CardProfileProps } from '.'

const props: CardProfileProps = {
  image: '/img/me.jpg',
  name: 'Leandro Lopes',
  age: 27,
  city: 'Santa Cruz do Sul - RS',
  liking: 'Programação, design, música, séries e filmes',
  dream:
    'Trabalhar com o que amo, utilizando meu conhecimento para criar aplicações que possam ajudar as pessoas.',
}

const { getByRole, getByText } = screen

describe('<CardProfile />', () => {
  it('should render CardProfile', () => {
    const { container } = renderWithTheme(<CardProfile {...props} />)

    const image = getByRole('img', { name: 'Leandro Lopes' })
    const name = getByText('Leandro Lopes')
    const age = getByText('27 anos')
    const city = getByText('Santa Cruz do Sul - RS')
    const liking = getByText('Programação, design, música, séries e filmes')
    const dream = getByText(
      'Trabalhar com o que amo, utilizando meu conhecimento para criar aplicações que possam ajudar as pessoas.',
    )

    expect(image).toHaveAttribute('src', '/img/me.jpg')
    expect(name).toBeInTheDocument()
    expect(age).toBeInTheDocument()
    expect(city).toBeInTheDocument()
    expect(liking).toBeInTheDocument()
    expect(dream).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
