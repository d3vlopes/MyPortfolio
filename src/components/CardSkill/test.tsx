import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'
import { Sword } from '@styled-icons/remix-fill/Sword'

import CardSkill from '.'
import mock from './mock'

const { getByTestId, getByRole, getByText } = screen

describe('<CardSkill />', () => {
  it('should render CardSkill', () => {
    renderWithTheme(
      <CardSkill
        icon={<Sword data-testid="icon" />}
        title={mock[0].title}
        description={mock[0].description}
      />,
    )

    const icon = getByTestId('icon')
    const title = getByRole('heading', { name: /desafios/i })
    const description = getByText(
      /Estou sempre em busca de novos desafios que me tirem do meu porto seguro./i,
    )

    expect(icon).toBeInTheDocument()
    expect(title).toBeInTheDocument()
    expect(description).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = renderWithTheme(
      <CardSkill
        icon={<Sword data-testid="icon" />}
        title={mock[0].title}
        description={mock[0].description}
      />,
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
