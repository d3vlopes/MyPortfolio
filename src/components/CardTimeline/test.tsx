import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import CardTimeline, { CardTimelineProps } from '.'

const props: CardTimelineProps = {
  title: 'CardTimeline',
  content: '<p>Loren ipsum dolor</p>',
}

const { getByRole, getByText } = screen

describe('<CardTimeline />', () => {
  it('should render CardTimeline', () => {
    const { container } = renderWithTheme(<CardTimeline {...props} />)

    const title = getByRole('heading', { name: 'CardTimeline' })
    const content = getByText('Loren ipsum dolor')
    const p = container.querySelector('p')

    expect(title).toBeInTheDocument()
    expect(content).toBeInTheDocument()
    expect(p?.tagName.toLowerCase()).toBe('p')
  })

  it('should match snapshot ', () => {
    const { container } = renderWithTheme(<CardTimeline {...props} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
