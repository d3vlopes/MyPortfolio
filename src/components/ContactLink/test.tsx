import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'
import { Props } from 'react-inlinesvg'

import ContactLink, { ContactLinkProps } from '.'

const { getByLabelText } = screen

const mock: ContactLinkProps = {
  icon: '/img/icon.svg',
  name: 'Github',
  link: 'www.github.com/user/name',
}

jest.mock('react-inlinesvg', () => {
  function ReactInlineSVG({ name }: Props) {
    return <svg aria-label={name} />
  }

  return ReactInlineSVG
})

describe('<ContactLinks />', () => {
  it('should render ContactLinks', () => {
    const { container } = renderWithTheme(<ContactLink {...mock} />)

    const icon = getByLabelText(mock.name)
    const a = container.querySelector('a')

    expect(icon).toBeInTheDocument()
    expect(a).toHaveAttribute('href', mock.link)
  })
})
