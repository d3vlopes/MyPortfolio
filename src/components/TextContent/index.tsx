import * as S from './styles'

export type TextContentProps = {
  content: string
}

const TextContent = ({ content }: TextContentProps) => {
  return (
    <S.Wrapper>
      <div
        className="content"
        dangerouslySetInnerHTML={{ __html: content }}
      ></div>
    </S.Wrapper>
  )
}

export default TextContent
