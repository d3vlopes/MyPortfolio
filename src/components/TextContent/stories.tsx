import { Meta, Story } from '@storybook/react'
import TextContent, { TextContentProps } from '.'
import mock, { project, code } from './mock'

export default {
  title: 'Components/TextContent',
  component: TextContent,
  args: {
    content: mock,
  },
} as Meta

export const Default: Story<TextContentProps> = (args) => {
  return (
    <div style={{ maxWidth: '59.7rem', margin: '0 auto' }}>
      <TextContent {...args} />
    </div>
  )
}

export const Project: Story<TextContentProps> = () => {
  return (
    <div style={{ maxWidth: '59.7rem', margin: '0 auto' }}>
      <TextContent content={project} />
    </div>
  )
}

export const Blog: Story<TextContentProps> = () => {
  return (
    <div style={{ maxWidth: '59.7rem', margin: '0 auto' }}>
      <TextContent content={code} />
    </div>
  )
}
