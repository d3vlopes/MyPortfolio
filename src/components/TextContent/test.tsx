import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import TextContent from '.'

const { getByRole } = screen

describe('<TextContent />', () => {
  it('should render TextContent', () => {
    const { container } = renderWithTheme(
      <TextContent content="<h6>Children</h6>" />,
    )

    const h6 = container.querySelector('h6')

    expect(h6?.tagName.toLowerCase()).toBe('h6')
    expect(getByRole('heading', { name: 'Children' })).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = renderWithTheme(
      <TextContent content="<h6>Children</h6>" />,
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
