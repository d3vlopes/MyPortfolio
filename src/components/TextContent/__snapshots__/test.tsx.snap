// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextContent /> should match snapshot 1`] = `
.c0 {
  font-size: 2.2rem;
  line-height: 180%;
  color: #DEE7EA;
}

.c0 h1,
.c0 h2,
.c0 h3,
.c0 h4,
.c0 h5,
.c0 h6 {
  font-family: Gilroy-Bold;
  color: #FFFFFF;
  margin: 1.6rem 0;
}

.c0 p {
  margin-bottom: 1.6rem;
  max-width: 59.7rem;
}

.c0 a,
.c0 a:visited,
.c0 a:link {
  color: #6D59A8;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c0 a:hover {
  color: #543F8C;
}

.c0 a:hover:hover > img {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c0 img {
  max-width: 100%;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  margin: 1.6rem 0;
}

.c0 img:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c0 .image {
  line-height: 0;
  margin: 2.4rem 0;
}

.c0 .image figcaption {
  font-size: 1.6rem;
  padding: 1.6rem;
  text-align: center;
  line-height: 1.3;
}

.c0 .image-half {
  max-width: 50%;
}

.c0 .image-style-side {
  max-width: 50%;
  float: right;
  margin: 2.4rem 0;
}

.c0 ul,
.c0 ol {
  padding: 0 2.4rem 1.6rem;
}

.c0 hr {
  border: none;
  border-bottom: 0.1rem solid #3B3A51;
}

.c0 code {
  font-family: monospace;
  color: #DEE7EA;
  font-size: 1.6rem;
  background: #302F43;
  padding: 0.2rem;
  margin: 0.2rem;
}

.c0 pre {
  background: #302F43;
  padding: 2.4rem;
  font-family: monospace;
  color: #FFFFFF;
  margin: 2.4rem 0;
  width: 100%;
  overflow-x: auto;
  font-size: 1.6rem;
}

.c0 pre code {
  color: inherit;
  background: inherit;
}

.c0 .table {
  width: 100%;
  overflow: hidden;
  overflow-x: auto;
}

.c0 table {
  width: 100%;
  border-collapse: collapse;
  margin: 2.4rem 0;
}

.c0 table td,
.c0 table th {
  padding: 1.6rem;
  border-bottom: 0.1rem solid #3B3A51;
}

.c0 blockquote {
  border-left: 0.5rem solid #6D59A8;
  color: #DEE7EA;
  padding-left: 2.4rem;
  margin: 2.4rem;
}

@media (max-width:768px) {
  .c0 {
    font-size: 1.6rem;
  }
}

@media (max-width:768px) {
  .c0 img:hover {
    -webkit-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }
}

@media (max-width:768px) {
  .c0 .image-style-side {
    max-width: 100%;
    float: none;
  }
}

<div
  class="c0"
>
  <div
    class="content"
  >
    <h6>
      Children
    </h6>
  </div>
</div>
`
